src/containers/Transaction/Transaction.js

```diff
@@ -49,27 +60,44 @@ const Transaction = ({ stockExchangeData }) => {
     resolver: yupResolver(schema)
   });
 
-  const [isBuying, currencyName, amount] = [watch('isBuying'), watch('currencyName'), watch('amount')];
+  const history = useHistory();
+  const location = useLocation();
   const dispatch = useDispatch();
+  const { currency: currencyUrlParam } = useParams();
+  const [
+    isBuying, price, currencyName, amount
+  ] = [watch('isBuying'), watch('price'), watch('currencyName'), watch('amount')];
 
-  const onSubmit = (data) => {
-    const { amount, currencyName, isBuying, price } = data;
+  const isCurrencyAvailable = (name) => stockExchangeData.map((currency) => currency.name).includes(name);
```

Comment:
That combination iterates over array twice, why not simplify that?
```suggestion
  const isCurrencyAvailable = (name) => stockExchangeData.some((currency) => currency.name === name);
```